src/Raven.Server/Documents/DocumentIdWorker.cs

```diff
@@ -378,23 +380,25 @@ private static ByteStringContext.InternalScope UnicodeGetLowerIdAndStorageKey<TT
                     ThrowDocumentIdTooBig(str);
 
                 byte* id = buffer.Ptr + strLength * sizeof(char) + maxStrSize;
-                byte* writePos = id;
-                int idSize = Encoding.GetBytes(pChars, strLength, writePos + maxIdLenSize, maxStrSize);
+                int idSize = Encoding.GetBytes(pChars, strLength, id + maxIdLenSize, maxStrSize);
 
                 var actualIdLenSize = JsonParserState.VariableSizeIntSize(idSize);
                 if (actualIdLenSize < maxIdLenSize)
                 {
                     var movePtr = maxIdLenSize - actualIdLenSize;
                     id += movePtr;
-                    writePos += movePtr;
                 }
 
+                byte* writePos = id;
+                _jsonParserState.FindEscapePositionsAndEscapeControls(id + maxIdLenSize, ref idSize, escapePositionsSize);
```

Comment:
Main fix:
This change performs the escape operation directly on the final ID, ensuring that control characters (ASCII < 32) are properly escaped as part of the process.